### ISSUE TITLE
Conclusions

### ISSUE THREAD

**Issue by cavaglia** [PhoenixBirdCreations/IPAM2021_ML#19]
Anything else to add in conclusions?

**Comment by frenbox** [comment]
Random forest classifier is a decision tree based classifier where the decision rules (for classification) places specific cuts (conditions) on the input features. K-Nearest Neighbor, on the other hand, computes nearest neighbors in the training set around a specific point of interest in the parameter space of input features. 

We found that the Random Forest classifier performs better than KNN, as seen in the ROC curves, in the testing data set  of O2 GstLAL for both **HasNS** and **HasRemnant**. However on the O3 MDC dataset containing 3 additional CBC pipelines, RF showed more variation in ROC curve among the pipelines compared to KNN. The RF performance for **HasRemnant** in O3 GstLAL is consistent with O2 dataset but under-performs sightly for other 3 pipelines. **We think(In this case) that the decision tree based alogrithm (RF) is not equally portable across pipelines as much has KNN**. 

The nature of RF algorithm might be suitable for **HasNS**  classifier which benefits from learning right cuts (conditions) on the secondary mass feature. For **HasRemnant** algorithms tries to the Foucart equation curve that differentiates non zero with zero post Remnant merger mass. While both algorithms can learn this boundary (as  seen in O2 dataset), due to algorithmic nature, RF is less portable for other pipelines compared to KNN.

**Do we need to address the spirr gap for HasNS**

I am just writing my thoughts out loud. Need your help to summarize this and put it in conclusion.

**Comment by frenbox** [comment]
Do we need to comment on the drop in performance of **HasNS** for spiir?

Or write under investigation.

**Comment by miqmiq** [comment]
I would say it needs more investigation

**Comment by lmagana3** [comment]
I would just say a one or two line comment would be enough. But no strong opinion on this.

**Comment by cavaglia** [comment]
Please all take a look at the new conclusions.

**Comment by miqmiq** [comment]
I've just looked at them, they seem fine

**Comment by dtsenekl** [comment]
I read the conclusions and they look good to me too. I would rephrase  
the first sentence to be: GW CBC signals to be detected by the LVK  
detectors, as it is for the future observations. But I think it's a  
detail.


Quoting cavaglia ***@***.***>:
